client/app.jsx

```diff
@@ -126,10 +172,14 @@ class App extends React.Component {
               clearDates={this.clearDates.bind(this)}
             />
           }
-          <div className="sub-component">
-            <h3>Guests</h3>
-            <img className="icon" src={downArrow} />
-          </div>
+          {this.renderGuestTitle.call(this)}
+          {this.state.isSelectingGuests ? 
```

Comment:
I think you could refactor this to the following:

{this.state.isSelectingGuests && <Guests (pass props here) /> }

Just something to consider